src/System.IO.Compression/tests/Performance/Perf.DeflateStream.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using Microsoft.Xunit.Performance;
+using System.Collections.Generic;
+
+namespace System.IO.Compression.Tests
+{
+    public class Perf_DeflateStream
+    {
+        /// <summary>
+        /// Yields an Enumerable list of paths to GZTestData files
+        /// </summary>
+        public static IEnumerable<object[]> GZTestFiles()
+        {
+            yield return new object[] { "GZTestData/GZTestDocument.txt.gz" };
+            yield return new object[] { "GZTestData/GZTestDocument.docx.gz" };
+        }
+
+        [Benchmark]
+        [MemberData("GZTestFiles")]
+        public async void Decompress(string testFilePath)
+        {
+            int _bufferSize = 1024;
+            var bytes = new byte[_bufferSize];
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                int retCount = -1;
+                using (var gzStream = await LocalMemoryStream.readAppFileAsync(testFilePath))
+                using (var strippedMs = StripHeaderAndFooter.Strip(gzStream))
+                using (var zip = new DeflateStream(strippedMs, CompressionMode.Decompress))
+                using (iteration.StartMeasurement())
+                {
+                    while (retCount != 0)
+                    {
+                        retCount = zip.Read(bytes, 0, _bufferSize);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [InlineData(100)]
+        [InlineData(1000)]
+        public void Compress(int numberOfWrites)
+        {
+            var bytes = Text.Encoding.UTF8.GetBytes(PerfUtils.CreateString(100));
```

Comment:
Writing the same repeated short sequence of data could make the data quite compressible and thus impact the performance of the Compress operation.  You might want a few tests here, one for very compressible data (e.g. a few characters repeated over and over), one for very non-compressible data (e.g. cryptographically random generated content), and one somewhere in the middle.